### ISSUE TITLE
💅 `lint/nursery/noShadow` incorrectly flags scoped destructured variables

### ISSUE THREAD

**Issue by erunion** [biomejs/biome#6921]
### Environment information

```bash
:
  Version:                      2.1.2
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v20.19.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         npm/10.8.2

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.jsonc
  Formatter enabled:            false
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false

Workspace:
  Open Documents:               0
```

### Rule name

lint/nursery/noShadow

### Playground link

https://biomejs.dev/playground/?code=ZgB1AG4AYwB0AGkAbwBuACAAcwBvAG0AZQBGAHUAbgBjAHQAaQBvAG4AKABjAG8AbgBkAGkAdABpAG8AbgA6ACAAYgBvAG8AbABlAGEAbgApACAAewAKACAAIABpAGYAIAAoAGMAbwBuAGQAaQB0AGkAbwBuACkAIAB7AAoAIAAgACAAIABjAG8AbgBzAHQAIABzAHQAcgAgAD0AIAAnAHQAaABpAHMAIABpAHMAIABhACAAbQBlAHMAcwBhAGcAZQAnADsACgAgACAAIAAgAGMAbwBuAHMAdAAgAHsAIABzAHQAcgA6ACAAZABlAHMAdAByAHUAYwB0AHUAcgBlAGQAUwB0AHIAIAB9ACAAPQAgAHsAIABzAHQAcgA6ACAAJwB0AGgAaQBzACAAaQBzACAAYQAgAG0AZQBzAHMAYQBnAGUAJwAgAH0AOwAKACAAIAAgACAAYwBvAG4AcwB0ACAAbwBiAGoAUwB0AHIAIAA9ACAAewAgAHMAdAByADoAIAAnAHQAaABpAHMAIABpAHMAIABhACAAbQBlAHMAcwBhAGcAZQAnACAAfQA7AAoAIAAgACAAIAByAGUAdAB1AHIAbgAgAHsACgAgACAAIAAgACAAIABzAHQAcgAsAAoAIAAgACAAIAAgACAAZABlAHMAdAByAHUAYwB0AHUAcgBlAGQAUwB0AHIALAAKACAAIAAgACAAIAAgAG8AYgBqAFMAdAByAAoAIAAgACAAIAB9ADsACgAgACAAfQAKAAoAIAAgAGMAbwBuAHMAdAAgAHMAdAByACAAPQAgACcAdABoAGkAcwAgAGkAcwAgAGEAIABtAGUAcwBzAGEAZwBlACcAOwAKACAAIABjAG8AbgBzAHQAIAB7ACAAcwB0AHIAOgAgAGQAZQBzAHQAcgB1AGMAdAB1AHIAZQBkAFMAdAByACAAfQAgAD0AIAB7ACAAcwB0AHIAOgAgACcAdABoAGkAcwAgAGkAcwAgAGEAIABkAGkAZgBmAGUAcgBlAG4AdAAgAG0AZQBzAHMAYQBnAGUAJwAgAH0AOwAKACAAIABjAG8AbgBzAHQAIABvAGIAagBTAHQAcgAgAD0AIAB7ACAAcwB0AHIAOgAgACcAdABoAGkAcwAgAGkAcwAgAGEAIABkAGkAZgBmAGUAcgBlAG4AdAAgAG0AZQBzAHMAYQBnAGUAJwAgAH0AOwAKACAAIAByAGUAdAB1AHIAbgAgAHsACgAgACAAIAAgAHMAdAByACwACgAgACAAIAAgAGQAZQBzAHQAcgB1AGMAdAB1AHIAZQBkAFMAdAByACwACgAgACAAIAAgAG8AYgBqAFMAdAByAAoAIAAgAH0AOwAKAH0ACgAKAHMAbwBtAGUARgB1AG4AYwB0AGkAbwBuACgAdAByAHUAZQApADsACgAKAA%3D%3D&ruleDomains.project=all

### Expected result

The playground doesn't seem to be replicating this issue but locally on a fresh install I'm seeing that if a variable is scoped in a logic block and that variable was destructured then it gets picked up as having a shadow if it's reused outside of that logic block. Other variables that are reused within this same scoped flow do not get flagged as having shadows.

<img width="1019" height="574" alt="Image" src="https://github.com/user-attachments/assets/3874d126-6de5-4d71-b9cb-fb96f1d1a539" />

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
